pkg/autodetect/main.go

```diff
@@ -99,7 +101,15 @@ func (b *Background) requireUpdates(deps *appsv1.DeploymentList) []*appsv1.Deplo
 		} else {
 			// Try to update the sidecar if is required
 			jaeger := inject.Select(dep, instances)
-			updated := inject.UpdateSideCar(jaeger, dep)
+			if jaeger == nil {
+				log.WithFields(log.Fields{
+					"deploymentName":      dep.Name,
+					"deploymentNamespace": dep.Namespace,
+				}).Debug("no suitable jaeger for this instance, skipping injection")
+				return nil
```

Comment:
I think this should be a continue not return, we need to process the rest of the deployments.

Reply:
Changed.